pkg/controller/vitessshard/reconcile_disk.go

```diff
@@ -21,6 +21,11 @@ const (
 func (r *ReconcileVitessShard) reconcileDisk(ctx context.Context, vts *planetscalev2.VitessShard) (reconcile.Result, error) {
 	resultBuilder := &results.Builder{}
 
+	// If the user has specified their disk resizes to be handled externally, wait for a manual rollout to apply changes.
+	if *vts.Spec.UpdateStrategy.DataVolumeClaimResize == planetscalev2.ExternalDataVolumeClaimResizeType {
```

Comment:
Let's bail if it's anything but Immediate. We don't know what other types may be added later, so we should be conservative in when we enable this.

Also, if the main UpdateStrategy.Type is Immediate, we should go ahead and do this without even checking DataVolumeClaimResize. That way setting the overall type to Immediate really makes everything immediate.

Reply:
Sounds good, done.